src/rust/src/concurrent.rs

```diff
@@ -11,83 +11,157 @@ use extendr_api::prelude::*;
 use extendr_api::Conversions;
 use std::result::Result;
 
-// This is the standard way the main thread which can call the R session,
-// should process a request from a polars thread worker to run an R function
-fn serve_r((probj, s): (ParRObj, pl::Series)) -> Result<pl::Series, Box<dyn std::error::Error>> {
-    //unpack user-R-function
-    let f = probj.0.as_function().ok_or_else(|| {
-        extendr_api::error::Error::Other(format!(
-            "provided input is not an R function but a: {:?}",
-            probj.0
-        ))
-    })?;
+// define any possible signature of R lambdas
+#[derive(Debug)]
+pub enum RFnSignature {
+    //... read as function with input 1 Series mapped to output 1 Series
+    FnSeriesTOSeries(ParRObj, pl::Series),
```

Comment:
Is the `TO` on purpose or should it be `FnSeriesToSeries` instead?

Reply:
thx :)